core/jazz-web/UIAutomation/e2e/page-objects/jazzservices.po.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ *Copyright 2016-2017 T Mobile, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); You may not use
+ * this file except in compliance with the License. A copy of the License is located at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { browser, by, element, $, $$ } from 'protractor';
+const fs = require('fs');
+const applicationDomain = '';
+
+export class Jazz {
+
+  navigateToJazzGet() {
+    return browser.driver.get(applicationDomain);
+  }
+  
+  getCreateService() {
+    return element(by.xpath('//div[@class="page-title-wrap hide-for-mobile"]/btn-primary-with-icon[@text="CREATE SERVICE"]'));
+  }
+ 
+  getPageTitle() {
+    return element(by.xpath('//div/div[2]/div/services-list/section/navigation-bar/div'));
+  }
+
+ getFunction() {
+  
+    return element(by.xpath('//div[@class="service-box"]/p[text()="FUNCTION"]/parent::div/parent::section'));
+  }
+
+  getLambda() {
+  
+    return element(by.xpath('//div[@class="service-box"]/p[text()="LAMBDA"]/parent::div/parent::section'));
+  }
+
+  getWebsite() {
+  
+  return element(by.xpath('//div[@class="service-box"]/p[text()="WEBSITE"]/parent::div/parent::section'));
+  }
+
+  getAws() {
+
+      return element(by.xpath('/html/body/app-root/div/services/div/div[2]/div/services-list/sidebar/div/div[2]/create-service/div/div/section[2]/div/div[2]/section[1]/div/div/span[33]'));
```

Comment:
is this the best way of getting to the path. This is going to be a big maintenance headache. Is there any other better way?

Reply:
Most of the xpaths are customized only.This is the way we are getting the paths.Customized paths will not make a problem.